yagb-core/src/emulator/ppu/ppu-cgb.ts

```diff
@@ -345,6 +363,42 @@ export class PpuCgb extends PpuBase {
         this.updatePalette(this.paletteOB1, value);
     };
 
+    private bgpiRead: ReadHandler = (_) =>
+        this.bgpi;
+    private bgpiWrite: WriteHandler = (_, value) => {
+        this.bgpi = value & 0xff;
+    }
+
+    private bgpdRead: ReadHandler = (_) =>
+        (this.reg[reg.lcdc] & lcdc.enable) === 0 || this.mode !== ppuMode.draw ? this.cram[this.bgpi & 0x3f] : 0xff;
+    private bgpdWrite: WriteHandler = (_, value) => {
+        const address = this.bgpi & 0x3f;
+
+        if((this.bgpi & 0x80) > 0) {
```

Comment:
Auto increment should happen *after* writing. Also, I'd prefer `!==0` (or just `if (this.bgpi & 0x80)`) here, but that's just a matter of taste.